### PR TITLE
Bug 1619470: Add support for listening to hg.mozilla.org pulse messsages.

### PR DESCRIPTION
**Pull request by tomprince**
This is to support [Bug 1619470](https://bugzilla.mozilla.org/show_bug.cgi?id=1619470), running `ci-admin automatically in cloudops's Jenkins instance. (cc: @edunham)

@oremj This shares a bunch of code with #18 but I realized that that may need a bit more thought on the exact interface that should have between taskcluster and jenkins, where-as I think the interface here is fairly obvious.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -47,6 +49,34 @@ func main() {
...
+		cli.StringFlag{
+			Name:   "pulse-host",
+			Usage:  "Pulse host to connect to",
+			Value:  "",
+			EnvVar: "PULSE_HOST",
+		},
+		cli.StringSliceFlag{
+			Name:   "hgmo-repo",
+			Usage:  "hg.mozilla.org repo to listend to pushes from (can be used multiple times)",
+			Value:  &cli.StringSlice{"ci/ci-admin", "ci/ci-configuration"},
```
Comment (tomprince) [inline main.go]: Not sure if these should be hard-coded?

**Comment by tomprince** [comment]
(also, this is currently untested)

File: proxyservice/jenkins.go [hunk 2]
```diff
@@ -116,3 +116,16 @@ func (j *Jenkins) TriggerDockerhubJob(data *DockerHubWebhookData) error {
 	params.Set("RawJSON", string(rawJSON))
 	return j.TriggerJob(path, params)
 }
+
+func (j *Jenkins) TriggerHgJob(repoUrl string, head string, data *HgMessage) error {
+	rawJSON, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("Error marshaling data: %v", err)
+	}
+	path := path.Join("/job/hgmo/job", repoUrl)
```
Comment (oremj) [inline proxyservice/jenkins.go]: What will the `repoUrl` look like?

Comment (tomprince) [reply proxyservice/jenkins.go]: `repoUrl` was the wrong thing (thanks for catching this). I've replaced it with `repoPath` which will look like `ci/ci-admin` or `mozilla-central` or `releases/mozilla-beta`.

Comment (oremj) [reply proxyservice/jenkins.go]: We should validate it as well before calling jenkins. We should check that it only includes: `[A-z0-9-/]` and then we need to split that on `/` and add job between them all.

File: proxyservice/jenkins.go [hunk 3]
```diff
@@ -116,3 +116,21 @@ func (j *Jenkins) TriggerDockerhubJob(data *DockerHubWebhookData) error {
 	params.Set("RawJSON", string(rawJSON))
 	return j.TriggerJob(path, params)
 }
+
+func (j *Jenkins) TriggerHgJob(repoPath string, repoUrl string, head string, data *HgMessage) error {
+	matches := regexp.MustCompile("^([A-z0-9-]*)/([A-z0-9-]*)$").FindStringSubmatch(repoPath)
```
Comment (oremj) [inline proxyservice/jenkins.go:121]:
```suggestion
	matches := regexp.MustCompile("^([A-Za-z][A-Za-z0-9-]*)/([A-Za-z][A-Za-z0-9-]*)$").FindStringSubmatch(repoPath)
```

**Comment by tomprince** [comment]
Added some test (with the docker-specific parts in #23 that this is based on).

**Comment by oremj** [comment]
Would you mind rebasing this PR?

**Comment by oremj** [comment]
Adding @moz-jvehent for security review.

**Pull request merged**